src/torchmetrics/functional/classification/generalized_dice.py

```diff
@@ -0,0 +1,247 @@
+    zero_division: Optional[int] = None,
+    average: str = "samples",
+    threshold: float = 0.5,
+    top_k: Optional[int] = None,
+    num_classes: Optional[int] = None,
+    multiclass: bool = True,
+    multidim: bool = True,
+    ignore_index: Optional[int] = None,
+    **kwargs: Any,
+) -> Tensor:
+    r"""Computes the Generalized Dice Score (GDS) metric:
+
+    .. math::
+        \text{GDS}=\sum_{i=1}^{C}\frac{2\cdot\text{TP}_i}{(2\cdot\text{TP}_i+\text{FP}_i+\text{FN}_i)\cdot w_i}
+
+    Where :math:`\text{C}` is the number of classes and :math:`\text{TP}_i`, :math:`\text{FP}_i` and :math:`\text{FN}`_i
+    represent the numbers of true positives, false positives and false negatives for class :math:`i`, respectively.
+    :math:`w_i` represents the weight of class :math:`i`.
+
+    The reduction method (how the recall scores are aggregated) is controlled by the
+    ``average`` parameter. Accepts all inputs listed in :ref:`pages/classification:input types`.
+
+    Args:
+        preds: Predictions from model (probabilities, logits or labels).
+
+        target: Ground truth values.
+
+        weight_type: Defines the type of weighting to apply. Should be one of the following:
+
+            - ``'square'`` [default]: Weight each class by the squared inverse of its support,
+              i.e., the inverse of its squared volume - :math:`\frac{1}{(tp + fn)^2}`.
+            - ``'simple'``: Weight each class by the inverse of its support, i.e.,
+              the inverse of its volume - :math:`\frac{1}{tp + fn}`.
+            - ``None``: All classes are assigned unitary weight. Equivalent to dice score.
+
+        zero_division:
+            The value to use for the score if denominator equals zero. If set to None, the score will be 1 if the
+            numerator is also 0, and 0 otherwise. Defaults to None.
+
+        average:
+            Defines the reduction that is applied. Should be one of the following:
+
+            - ``'samples'`` [default]: Calculate the metric for each sample, and average the metrics
+              across samples (with equal weights for each sample).
+            - ``'none'`` or ``None``: Calculate the metric for each sample separately, and return
+              the metric for every sample.
+
+        threshold:
+            Threshold for transforming probability or logit predictions to binary (0,1) predictions, in the case
+            of binary or multi-label inputs. Default value of 0.5 corresponds to input being probabilities.
+
+        top_k:
+            Number of the highest probability or logit score predictions considered finding the correct label.
+            The default value (``None``) will be interpreted as 1.
+
+        num_classes:
+            Number of classes.
+
+        multiclass:
+            Determines whether the input is multiclass (if True) or multilabel (if False). Defaults to True.
+
+        multidim:
+            Determines whether the input is multidim or not. Defaults to True.
```

Comment:
```suggestion
        multiclass:
            Determines whether the input is multiclass (if True) or multilabel (if False).

        multidim:
            Determines whether the input is multidim or not.
```
duplicated from func signature